content/tutorials/hello-world-node.md

```diff
@@ -35,320 +36,459 @@ Node.js installation varies depending on your operating system.
+    "producer": "node ./bin/index.js -- producer",
+    "consumer": "node ./bin/index.js -- consumer"
   },
   "keywords": [],
-  "author": "fluvio <user@fluvio.io> (fluvio.io)",
-  "license": "Apache 2.0"
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@types/node": "^14.11.2",
+    "typescript": "^4.0.3"
+  },
+  "dependencies": {
+    "@fluvio/client": "^5.0.0"
+  }
 }
 ```
 
+{{< idea >}}
 
-### Add Fluvio Client library
-
-The client library exports Fluvio data streaming APIs to the Node App.
-
-Use npm install to add <a href="https://www.npmjs.com/package/@fluvio/client" target="_blank">@fluvio/client</a> to the project:
+Your directory should now contain the following:
 
 ```bash
-$ npm install @fluvio/client --save
-
-> @fluvio/client@2.0.2 install /Users/user/fluvio-node-app/node_modules/@fluvio/client
-> nj-cli build
-
-Updating crates.io index
-...
-Finished dev [unoptimized + debuginfo] target(s) in 59.80s
+$ tree . -L 2
+.
+├── node_modules
+│   ├── @fluvio
+│   ├── @types
+│   └── typescript
+├── package-lock.json
+├── package.json
+├── src
+│   └── index.ts
+└── tsconfig.json
 
-+ @fluvio/client@2.0.2
-added 1 package from 1 contributor and audited 1 package in 61.137s
-found 0 vulnerabilities
+5 directories, 4 files
 ```
-A dependency to @fluvio/client is added to package.json. 
 
-### Build "Hello Word" Streaming App
+{{< /idea >}}
 
-Fluvio client needs a [default profile](/docs/cli/profiles) to identify the location and the authorization token of the cluster. The file was generated during cluster setup and it is available for download in your <a href="https://app.fluvio.io" target="_blank">Fluvio Cloud</a> account.
 
-#### Create Producer
+### Build "Hello, World!" Streaming Terminal App
 
-Inside your node project, create a _src_ directory, and add a _produce.js_ file: 
+We're going to write a small application, `StreamingApp`, that we will use to build a streaming interface from the terminal's standard input.
 
-```bash
-$ tree -L 2
-.
-├── node_modules
-│   └── @fluvio
-├── package-lock.json
-├── package.json
-└── src
-    └── produce.js
-```
+The following changes are made to the _./src/index.ts_ file. [TL;DR](#the-final-streamingapp-file)
 
-##### Producer Code
+### Importing Dependencies
 
-Add the following code in the _produce.js_ file:
+```TypeScript
+// Import Fluvio class and library resources;
+import Fluvio, { FluvioAdmin, Options, TopicProducer, PartitionConsumer } from "@fluvio/client";
+
+// Use Node.js `readline` module to read from command line;
+import { createInterface, Interface } from "readline";
 
-```js
-const FluvioClient = require('@fluvio/client');
- 
-async function produceMessage() {
-  try {
-    const flvConnection = await FluvioClient.connect();
-    let replica = await flvConnection.replica("my-topic", 0);
-    let len = await replica.produce("test");
-    
-    console.log("OK: %d bytes sent", len);
-  } catch (e) {
-    console.log("error: ", e.msg());
-  }
-}
 
```

Comment:
Remove blank lines

Reply:
commit 91158f0 attempts to clean up blank lines; not sure if there is a formatter out there that we could use to automate this.